src/main/java/gov/usgs/wma/waterdata/collections/BaseController.java

```diff
@@ -1,9 +1,40 @@
 package gov.usgs.wma.waterdata.collections;
 
+import org.springframework.http.MediaType;
+import org.springframework.web.HttpMediaTypeNotAcceptableException;
+
 public abstract class BaseController {
+	protected final String JSON = MediaType.APPLICATION_JSON_VALUE.replaceAll(".*/", "");
+	protected final String XML = MediaType.APPLICATION_XML_VALUE.replaceAll(".*/", "");
+	protected final String CONTENT_TYPE_MESS = "Content type must be either json or xml";
+
+	protected String contentType = JSON;
+	protected final String contentTypeDesc = "Content format returned, default json";
 	protected String ogc404Payload = "{\"code\":\"404\", \"description\":\"Requested data not found\"}";
 
 	protected static final String HTTP_400_DESCRIPTION = "A query parameter has an invalid value.";
 	protected static final String HTTP_404_DESCRIPTION = "The requested data was not found.";
 	protected static final String HTTP_500_DESCRIPTION = "Unexpected error occurred.";
-}
\ No newline at end of file
+
+	protected void determineContentType(String mimeType) throws HttpMediaTypeNotAcceptableException {
+		// TODO: for now just check the query parameter. later 'Accept' header.
+		// Allow for application/json or json as values
+		String baseMimeType = mimeType == null ? null : mimeType.replaceAll(".*/", "");
+		if (baseMimeType == null || baseMimeType.trim().isEmpty() || baseMimeType.equalsIgnoreCase(JSON)) {
+			contentType = JSON;
+		} else if (baseMimeType.equalsIgnoreCase(XML)) {
+			contentType = XML;
+		} else {
+			throw new HttpMediaTypeNotAcceptableException(CONTENT_TYPE_MESS);
+		}
+	}
+
+	protected boolean contentIsJson() {
+		return JSON.equals(contentType);
+	}
+
+	protected boolean contentIsXml() {
+		return XML.equals(contentType);
+	}
+
+}
```

Comment:
Since this only relates to the TimeSeriesController, I don't think it belongs in this class.  Perhaps determineContentType shouldn't throw an exception?  It might just return a default and/or consider the http accept header.

Reply:
I would say this potentially applies to all controllers. Every api call should have a query param "f" to specify the content
type. Already working on NGWMN-2099 which is a new/data endpoint that will return both json and WaterML
and use this.  It does need to be more flexible. There are likelyto be endpoints that only return json and perhaps
only WaterML. Eventually, the callers should specify the content types accepted and whether to default or error.

See previous comment about the accept header